app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -89,88 +89,117 @@ class DrawActivity : AppCompatActivity() {
                     onEditText = { layerName -> viewModel.setCurrentText(layerName) },
                     onSubmit = { layerName ->
                         if (layerName != null) {
-                            viewModel.addLayer(layerName)
+                            viewModel.addToList(layerName)
                         } else viewModel.dialogCompleted()
                     },
                 )
-            } else if (!it.isDialogOpen) {
-                isAddDialogOpen = false
-            }
-            if (it.isLayerSheetOpen && !isLayerListViewSheetOpen) {
+            } else isAddDialogOpen = false
+            if (it.isLayerListViewSheetOpen && !isLayerListViewSheetOpen) {
                 isLayerListViewSheetOpen = true
-                viewModel.setLayerListViewState()
-                bottomSheetLambda = dialogUtils.layerListBottomSheet(context = this,
+                onLayerListUpdate = dialogUtils.layerListBottomSheet(context = this,
                     layerList = it.layerList,
-                    onDialogComplete = { viewModel.dialogCompleted() },
-                    onDelete = { index ->
-                        viewModel.deleteLayer(index)
-                        if (!viewModel.isLayerListEmpty()) {
-                            bottomSheetLambda?.invoke(index, null, DialogOperation.DELETE)
-                        }
+                    onSheetComplete = { viewModel.dialogCompleted() },
+                    onDelete = { targetLayer ->
+                        isDeleteDialogOpen = true
+                        viewModel.setDeleteDialogState()
+                        viewModel.setTargetString(targetLayer)
+                        dialogUtils.confirmationDialog(context = this@DrawActivity,
+                            title = this@DrawActivity.getString(R.string.deleteLayerDialogTitle),
+                            message = this@DrawActivity.getString(
+                                R.string.deleteLayerDialogBody, targetLayer
+                            ),
+                            onConfirm = {
+                                if (!viewModel.isListSizeZero()) {
```

Comment:
Also keep in mind, something like that is business logic. The view shouldn't be knowing the requirements of the deletion, only the viewmodel should. Pass the index off to the viewmodel and let it do the work. All the `viewModel.property` outside the viewmodel should tell you the lines are bleeding together

Reply:
Yeah, this was a not a very well thought out approach to how the user flow should be when I implemented it.

Reply:
Replaced with a check for when the delete icon is clicked.